agent/consul/rpc.go

```diff
@@ -205,10 +206,23 @@ CHECK_LEADER:
 
 	// Handle the case of a known leader
 	if remoteServer != nil {
+		// A no leader error can happen if the remote server loses
+		// leadership, so it's worth to retry for that case.
 		err := s.forwardLeader(remoteServer, method, args, reply)
+		if structs.IsErrNoLeader(err) {
+			goto RETRY
+		}
 		return true, err
+	} else {
+		// If we are leaving and don't know about the leader, we
```

Comment:
Is this comment accurate, i.e is leaving the only situation where when remoteServer == nil && this is not a leader? Can it happen during leader elections too?

Reply:
The "leaving" part of that comment is because we are checking the leave drain flag, which only gets set when we leave. I'm actually going to move this check up here in another commit as it seems a little more robust, there will be a comment added about why I did that so plz. stay tuned.